app/controllers/buildings_controller.ts

```diff
@@ -1,70 +1,83 @@
 import Building from '#models/building'
-import Campus from '#models/campus'
+import { showValidator, indexValidator, byCampusValidator } from '#validators/building'
 import { HttpContext } from '@adonisjs/core/http'
 
 export default class BuildingsController {
   /**
    * Display a list of resource
    */
-  async index({ response }: HttpContext) {
-    try {
-      const buildings = await Building.all()
-      if (buildings.length === 0) {
-        return response.status(400).json({ message: 'No buildings in database' })
-      }
-      return response.status(200).json(buildings)
-    } catch (error) {
-      return response
-        .status(500)
-        .json({ message: 'Failed to fetch buildings', error: error.message })
+  async index({ request, response }: HttpContext) {
+    const { page, limit, includeCampus } = await request.validateUsing(indexValidator)
+
+    const buildingsQuery = Building.query()
+    const message = 'List of buildings'
+
+    if (includeCampus) {
```

Comment:
in general i think that checking specifically `if (includeCampus !== undefined && includeCampus === true)` helps to avoid JS logic madness
![image](https://github.com/user-attachments/assets/d0972aca-74a8-434d-8cd2-1be4b27448be)

Reply:
@mini-bomba global config coming soon™